# Why Loops?

### Save time when you code. Do things over and over again!

### Print me all the odd numbers from 1 to 100

In [6]:
# 100-2 = 98
for ind in range(1, 100, 2):
    print(ind)

1
3
5
7
9
11
13
15
17
19
21
23
25
27
29
31
33
35
37
39
41
43
45
47
49
51
53
55
57
59
61
63
65
67
69
71
73
75
77
79
81
83
85
87
89
91
93
95
97
99


### Display all the multiples of 6 between 0 and 120.

In [7]:
# 126-6 = 120
for ind in range(0, 126, 6):
    print(ind)

0
6
12
18
24
30
36
42
48
54
60
66
72
78
84
90
96
102
108
114
120


### Display all numbers between 100 and 500.

In [10]:
# 500 = 501 - 1
for num in range(100, 501):
    print(num)

100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192
193
194
195
196
197
198
199
200
201
202
203
204
205
206
207
208
209
210
211
212
213
214
215
216
217
218
219
220
221
222
223
224
225
226
227
228
229
230
231
232
233
234
235
236
237
238
239
240
241
242
243
244
245
246
247
248
249
250
251
252
253
254
255
256
257
258
259
260
261
262
263
264
265
266
267
268
269
270
271
272
273
274
275
276
277
278
279
280
281
282
283
284
285
286
287
288
289
290
291
292
293
294
295
296
297
298
299
300
301
302
303
304
305
306
307
308
309
310
311
312
313
314
315
316
317
318
319
320
321
322
323
324
325
326
327
328
329
330
331
332
333
334
335
336
337
338
339
340
341
342
343
344
345
346
347
348
349


### Print me the multiples of 3 from 0 to 30.

In [8]:
# you find the end limit of a for loop by subtracting the ending value and the step counter. So in this case, 31-3 = 28
for ind in range(0, 31, 3):
    print(ind)

0
3
6
9
12
15
18
21
24
27
30


In [11]:
print("Abhinav is my best friend")
print("Abhinav is my best friend")
print("Abhinav is my best friend")
print("Abhinav is my best friend")

Abhinav is my best friend
Abhinav is my best friend
Abhinav is my best friend
Abhinav is my best friend


In [12]:
for ind in range(10):
    print("Abhinav is my favortie student")

Abhinav is my favortie student
Abhinav is my favortie student
Abhinav is my favortie student
Abhinav is my favortie student
Abhinav is my favortie student
Abhinav is my favortie student
Abhinav is my favortie student
Abhinav is my favortie student
Abhinav is my favortie student
Abhinav is my favortie student


In [13]:
for ind in range(1000):
    print("a", end=' ')

a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a 

### How do you tell if a number is divisible by 2?

You check if it is divisble by 2

If the remainder of a number divided by 2 is 0. then it is divisible by 2. Or else, it is not.

We use the ```%``` sign to find the remainder of a division.

In [14]:
44 % 2

0

In [15]:
44/2

22.0

If you want to check if 2 things are the same, you use ```==``` sign

A single ```=``` sign is used to ```assign``` value to something

In [16]:
45 % 2 == 1

True

### If the remainder of a number when divided by 2 is 0, it is even. If the remainder of a number when divided by 2 is 1, it is odd.

In [17]:
for ind in range(1, 21):
    if ind % 2 == 0:
        print(ind, "is even")
    elif ind % 2 == 1:
        print(ind, "is odd")

1 is odd
2 is even
3 is odd
4 is even
5 is odd
6 is even
7 is odd
8 is even
9 is odd
10 is even
11 is odd
12 is even
13 is odd
14 is even
15 is odd
16 is even
17 is odd
18 is even
19 is odd
20 is even


### For all numbers between 1 and 250, print "STRIKE" if the number is odd, and "BOWL" if the number is even

In [18]:
for ind in range(1, 251):
    if ind % 2 == 0:
        print("BOWL")
    else:
        print("STRIKE")

STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRIKE
BOWL
STRI